### PR TITLE
Handle where_multi and returning total_found

### PR DESCRIPTION
**Pull request by sc0ttkclark**
https://central.tri.be/issues/120395

This handles implementation of where_multi() / found() for Attendees Table class.

It removes some search keys which are not yet available. We can revisit them before doing the final merge on this.

It adds some new search keys for Attendee Table to support as well.

It cleans up the schema definition for the Attendees Repository class so it can support even more lookups and also removes methods that are now no longer necessary because of the simple meta schema usage.

### CODE REVIEW

File: src/Tribe/Attendees_Table.php [hunk 1]
```diff
@@ -814,27 +814,78 @@ protected function attendee_reference( $reference ) {
...
+			'return_total_found' => true,
+		];
 
-		$search = isset( $_REQUEST['s'] ) ? esc_attr( trim( $_REQUEST['s'] ) ) : false;
-		if ( ! empty( $search ) ) {
-			$items = $this->filter_attendees_by_string( $search, $items );
+		$event_id = 0;
+
+		if ( ! empty( $_GET['event_id'] ) ) {
+			$event_id = absint( $_GET['event_id'] );
```
Comment (Camwyn) [inline src/Tribe/Attendees_Table.php]: maybe combine them? 
`$event_id = empty( $_GET['event_id'] ) ? 0 : absint( $_GET['event_id'] );`

File: src/Tribe/Attendees_Table.php [hunk 2]
```diff
@@ -814,27 +814,78 @@ protected function attendee_reference( $reference ) {
...
 		}
 
-		$total_items = count( $items );
+		$search = null;
 
-		$current_page = $this->get_pagenum();
-		$per_page = $this->get_items_per_page( $this->per_page_option );
-		$this->items  = array_slice( $items, ( $current_page - 1 ) * $per_page, $per_page );
+		if ( ! empty( $_REQUEST['s'] ) ) {
+			$search = sanitize_text_field( $_REQUEST['s'] );
```
Comment (Camwyn) [inline src/Tribe/Attendees_Table.php]: If you use a ternary here you don't have to nest the ifs:
`$search = ! empty( $_REQUEST['s'] ) ? sanitize_text_field( $_REQUEST['s'] ) : 0;`

Then you can pull out the `if ( ! empty( $search ) )`

File: src/Tribe/Tickets.php [hunk 3]
```diff
@@ -968,12 +968,23 @@ public function maybe_add_front_end_tickets_form( $content ) {
...
+		 *      @type array   $where_multi        List of ORM->where_multi() filters to use. [[what,args...]] format.
+		 * }
 		 *
-		 * @return array
+		 * @return array List of attendees, or an array of total_found and attendees if $args['return_total_found'].
 		 */
 		public static function get_event_attendees( $post_id, $args = [] ) {
-			$attendees = array();
+			$total_found = 0;
+			$attendees   = array();
```
Comment (Camwyn) [inline src/Tribe/Tickets.php]: `[]`

**Comment by sc0ttkclark** [comment]
I'm going to move this into it's own abstracted method to make it easier

**Pull request merged**